src/store/reducers/auth.js

```diff
@@ -0,0 +1,77 @@
+import { combineReducers, reduceReducers } from 'redux-loop';
+
+import requestData from '../transducers/requestData.js';
+import on from '../transducers/on.js';
+import fallback from '../transducers/fallback.js';
+import consequence from '../transducers/consequence.js';
+
+import OK from '../actions/OK.js';
+import ERR from '../actions/ERR.js';
+import {
+  login as loginAction,
+  saveToken as saveTokenAction,
+  initializeTokenFromStorage as initTokenAction,
+  invalidateToken as invalidateTokenAction,
+} from '../actions/auth.js';
+
+import { login } from '../../service/auth.js';
+import { storage } from '../../util';
+
+const findTokenString = RegExp.prototype.test.bind(/^sails\.sid/);
+
+export default combineReducers({
+  authorized: reduceReducers(
+    // Request login
+    requestData(loginAction.type, login, () => false, () => false),
+
+    // ACTION: login[OK] -> ACTION: saveToken
+    consequence(OK(loginAction.type), saveTokenAction, (_, res) => {
```

Comment:
Are `on` transducer for synchronous reducers and `consequence` for asynchronous reducers?

Reply:
Should also store the user information returned from the same login request.
![image](https://user-images.githubusercontent.com/7604366/49345335-67429200-f651-11e8-9610-34e3e4fa2fde.png)

Reply:
> Are on transducer for synchronous reducers and consequence for asynchronous reducers?

There is a big difference between `on` and `consequence`. `on` can mutate the store, while `consequence` can't do it.

`on`: taking an action, then updating store.
`consequence`: taking an action, then dispatching another action.

So the action `saveToken` will be dispatched after `OK(login)` because we need to save token after logged in.

> Should also store the user information returned from the same login request.

I agree and It's being planned. I'd like to open a separate pr to implement it. This one is about user authorization so let's keep it simple.

Reply:
Thank you for your clear explanation!